### PR TITLE
Running CI for testdata

### PR DESCRIPTION
**Pull request by robertbastian**
Hasn't happened since we excluded it from the workspace, and I don't want to fix issues at release time.

### CODE REVIEW

File: tools/make/data.toml [hunk 1]
```diff
@@ -78,6 +78,20 @@ args = [
     "--bin=make-testdata-legacy",
 ]
 
+[tasks.testdata-legacy]
+description = "Verify the data for the legacy icu_testdata crate"
+category = "ICU4X Data"
+dependencies = ["testdata-legacy-gen"]
+script_runner = "@duckscript"
+script = """
+    exec --fail-on-error cargo clean # free up space
```
Comment (sffc) [inline tools/make/data.toml:87]: not sure we want to automatically `cargo clean`

File: provider/datagen/src/baked_exporter.rs [hunk 2]
```diff
@@ -631,9 +631,11 @@ impl BakedExporter {
             quote! {
                 #(
                     #[macro_use]
+                    #features
                     #[path = #file_paths]
                     mod #mod_idents;
                     #[doc(inline)]
+                    #features
```
Comment (sffc) [inline provider/datagen/src/baked_exporter.rs:638]: is this a separate bug fix?

Comment (robertbastian) [reply provider/datagen/src/baked_exporter.rs:638]: yep

File: tools/make/data.toml (hunk 1 above)
Comment (robertbastian) [reply tools/make/data.toml]: reusing it instead

**Review by sffc — APPROVED** [review]
Nit: Please change the commit message to say that you're also fixing a bug with `#features` in the baked exporter

File: provider/datagen/src/baked_exporter.rs (hunk 2 above)
Comment (robertbastian) [reply provider/datagen/src/baked_exporter.rs]: removed

**Review by sffc — DISMISSED** [review]

**Review by Manishearth — APPROVED** [review]

**Review by sffc — COMMENTED** [review]
.

**Pull request merged**